portal-api/src/v1/api.js

```diff
@@ -268,33 +267,43 @@ const saveUtilisationReport = async (reportData, month, year, user, fileInfo) =>
   }
 };
 
-const getUtilisationReports = async (bankId, month, year) => {
+/**
+ * Gets utilisation reports for a specific bank. If a report
+ * period is not provided, all reports for that bank are
+ * returned. If a report period is provided, the report
+ * submitted for that report period is returned. Returned
+ * reports are ordered by year descending and month
+ * ascending.
+ * @param {string} bankId
+ * @param {import('../types/utilisation-reports').ReportPeriod} [reportPeriod]
+ */
+const getUtilisationReports = async (bankId, reportPeriod) => {
   try {
     if (!isValidBankId(bankId)) {
       console.error('Get utilisation reports failed with the following bank ID %s', bankId);
       throw new Error('Invalid bank ID provided: %s', bankId);
     }
 
-    if (month && !isValidMonth(parseInt(month, 10))) {
-      console.error('Get utilisation reports failed with the following month %s', month);
-      throw new Error('Invalid month provided: %s', month);
+    if (!reportPeriod) {
+      const response = await axios({
+        method: 'get',
+        url: `${DTFS_CENTRAL_API_URL}/v1/bank/${bankId}/utilisation-reports`,
+        headers: headers.central,
+      });
+      return response.data;
     }
 
-    if (year && !isValidYear(parseInt(year, 10))) {
-      console.error('Get utilisation reports failed with the following year %s', year);
-      throw new Error('Invalid year provided: %s', year);
+    if (!isValidReportPeriod(reportPeriod)) {
+      console.error('Get utilisation reports failed with the following report period:', reportPeriod);
+      throw new Error('Invalid report period provided: %s', reportPeriod);
     }
+    const { month, year } = reportPeriod.start;
 
     const response = await axios({
       method: 'get',
-      url: `${DTFS_CENTRAL_API_URL}/v1/bank/${bankId}/utilisation-reports`,
-      params: {
-        month,
-        year,
-      },
+      url: `${DTFS_CENTRAL_API_URL}/v1/bank/${bankId}/utilisation-reports?month=${month}&year=${year}`,
```

Comment:
When using axios we shouldn't ever need to manually construct query params, but instead pass them as the `params` option. As well as preventing the need to manually construct it, this will also prevent the need for us to URL encode values (if required).

Taking all these changes into account this function can be simplified to
```
const getUtilisationReports = async (bankId, reportPeriod) => {
  try {
    if (!isValidBankId(bankId)) {
      console.error('Get utilisation reports failed with the following bank ID %s', bankId);
      throw new Error(`Invalid bank ID provided: ${bankId}`);
    }

    if (reportPeriod && !isValidReportPeriod(reportPeriod)) {
      console.error('Get utilisation reports failed with the following report period:', reportPeriod);
      throw new Error(`Invalid report period provided: ${reportPeriod}`);
    }

    const response = await axios.get(`${DTFS_CENTRAL_API_URL}/v1/bank/${bankId}/utilisation-reports`, {
      params: reportPeriod ? reportPeriod.start : undefined,
      headers: headers.central,
    });
    return response.data;
  } catch (error) {
    console.error('Unable to get previous utilisation reports %s', error);
    throw error;
  }
};
```

Reply:
My reasoning was based on the pattern that `query` values are supposed to act as filters for the data, which is what they would be doing here. I also think you can bypass the URL encoding (to some extent) by storing then in a `query` object and then passing that to the `axios.get` call:
```
axios.get([url], {
  headers: headers.central,
  query: reportPeriod ? reportPeriod.start : undefined,
}
```

Reply:
Perfect, that's what I meant 😅